### ISSUE TITLE
Analytics Implementation or QA Support for OPIA Choose

### ISSUE THREAD

**Issue by Hampshire100** [department-of-veterans-affairs/va.gov-team#15793]
# Google Analytics Implementation or QA Request

## Description
- Is this request for Google Analytics implementation or QA?

Google Analytics Implementation

- Who is this request for?

    - [joshua.tuscher@va.gov, stephen.walker4@va.gov; alyson.glick@va.gov]
    
- What team/project is this work for?

    - [OPIA ChooseVA contract team]
    
- Provide the Domain where Google Analytics needs to be implemented

    No PII or PHI will be passed. 

    - [www.choose.va.gov]

- Please provide any upcoming deadlines or important milestones

    
    - [none really but it would be nice to have it up and configured and working before 11/11]

- Typical Website Interactions to Track

    - *Conversions - Understanding the User Flow:* _Provide URLs that are part of the application from introduction page to successful form completion; conditional pages that are visible depending on an applicant's answer should be marker accordingly._
    
    - [https://www.va.gov/va-welcome-kit-color.pdf]
    
    - *Other User Interactions*: _Provide details on other interactions and metrics, like click to help text links, that are helpful in understanding user behavior._
    
    - [PDF downloads, email submit]

- Certain conditions/responses that need to be reviewed

    - [unknown]

- Additional Comments

    - [None. ]

> Please leave the following blank

## Acceptance Criteria
- [ ] QA has been completed
- [ ] VFS team completes brief [VSP Collaboration Cycle Feedback](https://adhoc.optimalworkshop.com/questions/20260uu8-0-0/questions/before) survey

## Definition of Done
- [ ] All appropriate issue tagging is completed
- [ ] All AC completed

**Comment by jonwehausen** [comment]
Hi @Hampshire100 , confirming receipt

**Comment by bsmartin-ep** [comment]
Hello @Hampshire100 -

Looks like joshua.tuscher@va.gov already has access.  They should just need to reset their password.

stephen.walker4@va.gov and alyson.glick@va.gov have been added and should receive an email shortly.

I granted you _Read and Analyze_ access.  If you need more access for things like custom segments and dashboards, please let me know.

Thanks!

**Comment by jonwehausen** [comment]
Thanks @bsmartin-ep , reopening as this is in-fact an implementation request. 

@Hampshire100, as a first step here -- we will need to get our GTM snippet deployed on the choose.va.gov landing page: 

### [_Only if there is a staging version of the page_], staging GTM snippet: 

Paste this code as high in the `<head>` of the page as possible:
```javascript
<!-- Google Tag Manager -->
<script>(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
'https://www.googletagmanager.com/gtm.js?id='+i+dl+ '&gtm_auth=inC4EKQce9vlWpRVcowiyQ&gtm_preview=env-661&gtm_cookies_win=x';f.parentNode.insertBefore(j,f);
})(window,document,'script','dataLayer','GTM-WFJWBD');</script>
<!-- End Google Tag Manager -->
```
Additionally, paste this code immediately after the opening `<body>` tag:
```javascript
<!-- Google Tag Manager (noscript) -->
<noscript><iframe src="https://www.googletagmanager.com/ns.html?id=GTM-WFJWBD&gtm_auth=inC4EKQce9vlWpRVcowiyQ&gtm_preview=env-661&gtm_cookies_win=x"
height="0" width="0" style="display:none;visibility:hidden"></iframe></noscript>
<!-- End Google Tag Manager (noscript) -->
```
------------------------------------------------------------------------------

### GTM snippet for choose.va.gov
Paste this code as high in the `<head>` of the page as possible:
```javascript
<!-- Google Tag Manager -->
<script>(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
'https://www.googletagmanager.com/gtm.js?id='+i+dl+ '&gtm_auth=XWK-rHkcb1wp4DCOCzhPiQ&gtm_preview=env-1&gtm_cookies_win=x';f.parentNode.insertBefore(j,f);
})(window,document,'script','dataLayer','GTM-WFJWBD');</script>
<!-- End Google Tag Manager -->
```
Additionally, paste this code immediately after the opening `<body>` tag:
```javascript
<!-- Google Tag Manager (noscript) -->
<noscript><iframe src="https://www.googletagmanager.com/ns.html?id=GTM-WFJWBD&gtm_auth=XWK-rHkcb1wp4DCOCzhPiQ&gtm_preview=env-1&gtm_cookies_win=x"
height="0" width="0" style="display:none;visibility:hidden"></iframe></noscript>
<!-- End Google Tag Manager (noscript) -->
```

----------------------------------------------------------------------------------------------------
### Sign-Up Form Tracking
- In order to collect the volume of submissions of the "Learn More" signup form, we would like to propose the following data layer code be implemented per specification:
- Please notice the double `--` before `submission` and the single `-` prior to `successful` or `failed`
- With each `-successful` and `-failed` event, we would like to include the metadata of interested benefits, passing a boolean of `true` or `false` for each checkbox that is clicked
- There aren't any other components of the form we can track without PII risk
 
Description of Interaction | Data Layer Code
------------ | -------------
**_All_** submissions of the form (successful or failed) | `dataLayer.push({`<br>`'event': 'choose-va-learn-more--submission'});`
**_Successful_** submissions of the form | `dataLayer.push({`<br>`'event': 'choose-va-learn-more--submission-successful'`<br>`'health-care-benefit-interest': true\|false`<br>`'disability-compensation-benefit-interest': true\|false`<br>`'education-benefit-interest': true\|false`<br>`'other-benefit-interest': true\|false`<br>`});`
**_Failed_** submissions of the form | `dataLayer.push({`<br>`'event': 'choose-va-learn-more--submission-failed'`<br>`'health-care-benefit-interest': true\|false`<br>`'disability-compensation-benefit-interest': true\|false`<br>`'education-benefit-interest': true\|false`<br>`'other-benefit-interest': true\|false`<br>`});`

### Welcome Kit Downloads
- Once we have our GTM snippet deployed on the page, we should automatically begin capturing downloads of the welcome kit per our global PDF download tracking

### Campaign UTM Tagging
- @Hampshire100 we would like to learn more about your campaign strategy here, as we would like to work with you all to establish a UTM tagging structure to support the different mediums you'll be campaigning -- let us know if this is already something you and your team have an agenda for. 

cc: @joanneesteban @bsmartin-ep @bmcgrady-ep

**Comment by jonwehausen** [comment]
Hi @Hampshire100 , 

Wanted to follow up with you here -- did you all have any further questions regarding the implementation?

**Comment by bsmartin-ep** [comment]
Placeholder GTM workspace for this:

https://tagmanager.google.com/#/container/accounts/144676348/containers/2214120/workspaces/1000392

**Comment by jonwehausen** [comment]
12/2 Sent a follow-up email requesting next steps

**Comment by joanneesteban** [comment]
Moving to blocked until we hear back from Josh.

**Comment by jonwehausen** [comment]
## Brief Notes from meeting 12/16
- Overall goal of the campaign is to increase awareness and drive traffic usage to  veteran benefits
- Existing Choose.va.gov campaign will run through the end of January with the implied expectation it will continue into 2021
- UTM taxonomy in place is thoroughly in place and consistent

## Next Steps
- Unblocking after meeting with Josh and his team 
- Validate GTM snippet deployment

**Comment by jonwehausen** [comment]
## 1/5/2021 Update

Reached out to Josh via email indicating to move forward with implementing the data layer for tracking form submissions.

**Comment by jonwehausen** [comment]
## 1/26/2021 Update
- Sent another follow-up to to Josh on remaining items for implementation

> 
> Hi Josh, 
> 
> Hope you're having a great start to 2021. 
> 
> I wanted to reach back out here with the goal of closing #15793 and begin capturing data for choose.va.gov.
> 
> I looked back at vaww.w2k8.internet.staging.va.gov/choose and the pageview tags look to be firing great. We will next want to have the GTM snippet with the appropriate environment variable deployed to production. 
> 
> Paste this code as high in the <head> of the page as possible:  
>```javascript
> 
> <script>(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
> new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
> j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
> 'https://www.googletagmanager.com/gtm.js?id='+i+dl+ '&gtm_auth=XWK-rHkcb1wp4DCOCzhPiQ&gtm_preview=env-1&gtm_cookies_win=x';f.parentNode.insertBefore(j,f);
> })(window,document,'script','dataLayer','GTM-WFJWBD');</script>
> 
>```
>```javascript
> Additionally, paste this code immediately after the opening <body> tag:
> 
> <noscript><iframe src="https://www.googletagmanager.com/ns.html?id=GTM-WFJWBD&gtm_auth=XWK-rHkcb1wp4DCOCzhPiQ&gtm_preview=env-1&gtm_cookies_win=x"
> height="0" width="0" style="display:none;visibility:hidden"></iframe></noscript>
> 
> 
>```
> From there, if desired, we can move forward with implementing the data layer specification found here to begin tracking form submissions. 
> 
> Thanks again here Josh, look forward to hearing from you. 
> 
> Best,
> Jon Wehausen

**Comment by jonwehausen** [comment]
### Update 3/3

Spoke with Josh Tuscher via email and confirmed we are now collecting data for choose.va.gov. 

Josh has chosen not to implement the form tracking data layer. 

Closing this issue.
